tests/test_cli/test_generate_all_protocols.py

```diff
@@ -0,0 +1,267 @@
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Test `generate-all-protocols` command."""
+
+import logging
+import pprint
+import shutil
+import subprocess
+from pathlib import Path
+from typing import Any
+from unittest import mock
+
+import pytest
+
+from aea.configurations.constants import (
+    DEFAULT_PROTOCOL_CONFIG_FILE,
+    DEFAULT_README_FILE,
+)
+from aea.exceptions import enforce
+from aea.manager.helpers import AEAProject
+from aea.test_tools.exceptions import AEATestingException
+from aea.test_tools.test_cases import AEATestCaseMany
+
+
+def _run_cli_patch(*args: Any, **kwargs: Any) -> None:
+
+    if "isort" in args or "black" in args:
+        return
+
+    print(f"Calling command {args} with kwargs {kwargs}")
+    return_code = subprocess.check_call(args, **kwargs)  # nosec
+    enforce(
+        return_code == 0,
+        f"Return code of {pprint.pformat(args)} is {return_code} != 0.",
+    )
+
+
+class TestGenerateAllProtcols(AEATestCaseMany):
+    """Test generate all protocols."""
+
+    use_packages_dir: bool = True
+    find_packages_patch: Any
+    package_path: Path
+
+    @classmethod
+    def setup_class(cls) -> None:
+        """Setup test class."""
+        super().setup_class()
+
+        cls.package_path = cls.t / "packages" / "fetchai" / "protocols" / "fipa"
+        cls.find_packages_patch = mock.patch(
+            "aea.cli.generate_all_protocols.find_protocols_in_local_registry",
+            return_value=(cls.package_path,),
+        )
+
+    def test_run(self, caplog) -> None:
+        """Test command invocation."""
+
+        with self.find_packages_patch, mock.patch.object(
+            AEAProject, "run_cli", new=_run_cli_patch
+        ), caplog.at_level(logging.INFO):
+            result = self.run_cli_command(
+                "generate-all-protocols",
+                str(self.packages_dir_path),
+                "--root-dir",
+                str(self.packages_dir_path.parent),
+            )
+
+            assert result.exit_code == 0
+            assert f"Processing protocol at path {self.package_path}"
+
+    def test_check_clean_pass(self) -> None:
+        """Test command invocation."""
+
+        with self.find_packages_patch, mock.patch.object(
+            AEAProject, "run_cli", new=_run_cli_patch
+        ), mock.patch(
+            "aea.cli.generate_all_protocols.check_working_tree_is_dirty",
+            return_value=True,
+        ):
```

Comment:
declare patch outside of the test and invoke it here only. It is repeated in the test below